07-http_gordle/final/internal/handlers/newgame/newgame_test.go

```diff
@@ -0,0 +1,76 @@
+package newgame
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"learngo-pockets/httpgordle/api"
+	"learngo-pockets/httpgordle/internal/domain"
+)
+
+func TestHandler(t *testing.T) {
+	corpusPath = "testdata/corpus.txt"
+	idFinderRegexp := regexp.MustCompile(`.+"id":"(\d+)".+`)
+
+	tt := map[string]struct {
+		wantStatusCode int
+		wantBody       string
+		creator        gameCreator
+	}{
+		"nominal": {
+			wantStatusCode: http.StatusCreated,
+			wantBody:       `{"id":"123456","attempts_left":5,"guesses":[],"word_length":5,"status":"Playing"}`,
+			creator: gameCreatorStub{
+				err: nil,
+			},
+		},
+	}
+
+	for name, testCase := range tt {
+
+		t.Run(name, func(t *testing.T) {
+			f := Handler(testCase.creator)
+
+			req, err := http.NewRequest(http.MethodPost, api.NewGamePath, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
+			rr := httptest.NewRecorder()
+
+			f.ServeHTTP(rr, req)
+
+			// Check the status code is what we expect.
+			assert.Equal(t, testCase.wantStatusCode, rr.Code)
+
+			// Check the response body is what we expect. Use JSONEq rather than Equal.
+			if testCase.wantBody != "" {
```

Comment:
This long block would call for a  `TestHandler_Success` and eventually `TestHandler_Failure`

Or, we could simply 
```go
if testCase.wantBody == "" {
  return
}

// replace the ID
//...
```